Cannot find module 'file://xxx'  when vite.config.ts import absolute path

### Describe the bug

when import absolute path  in   vite.config.ts vite@3.0.5 , it thrown an Error   Cannot find module 'file://xxx', but it  workwell in vite@3.0.4


<img width="1056" alt="image" src="https://user-images.githubusercontent.com/4457572/184056294-adc52b69-7c3d-4fbe-8087-f2e80e43d866.png">


### Reproduction

import absolute path  in   vite.config.ts 

### System Info

```shell
System:
    OS: macOS 12.2
    CPU: (8) arm64 Apple M1 Pro
    Memory: 115.59 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.79
    Firefox: 103.0.2
    Safari: 15.3
  npmPackages:
    @vitejs/plugin-legacy: ^1.4.3 => 1.8.2
    @vitejs/plugin-vue2: ^1.1.2 => 1.1.2
    vite: ^3.0.5 => 3.0.5
```


### Used Package Manager

yarn

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vite/src/node/config.ts
+++ b/packages/vite/src/node/config.ts
@@ -980,7 +980,7 @@ async function bundleConfigFile(
               if (path.relative(idPkgDir, fileName).startsWith('..')) {
                 return {
                   // normalize actual import after bundled as a single vite config
-                  path: pathToFileURL(idFsPath).href,
+                  path: isESM ? pathToFileURL(idFsPath).href : idFsPath,
                   external: true
                 }
               }

